raiden/network/transport/matrix/transport.py

```diff
@@ -550,24 +553,35 @@ def send_global(self, room: str, message: Message) -> None:
 
         assert self._global_rooms.get(room_name), f'Unknown global room: {room_name!r}'
 
-        def _send_global():
-            text = JSONSerializer.serialize(message)
-            room = self._global_rooms[room_name]
-            self.log.debug(
-                'Send global',
-                room_name=room_name,
-                room=room,
-                data=text.replace('\n', '\\n'),
-            )
-            room.send_text(text)
-
-        greenlet = self._spawn(_send_global)
-        greenlet.name = (
-            f'MatrixTransport.send_global '
-            f'node:{pex(self._raiden_service.address)} '
-            f'room:{room} '
-            f'user_id:{self._user_id}'
-        )
+        self._global_send_queue.put((room_name, message))
+
+    def _global_send_worker(self):
+        while True:
+            if self._stop_event.is_set():
+                break
+            if self._global_send_queue.qsize() > 0:
+                room_name, message = self._global_send_queue.get()
+
+                def _send_global(room_name, message):
+                    text = JSONSerializer.serialize(message)
+                    room = self._global_rooms[room_name]
+                    self.log.debug(
+                        'Send global',
+                        room_name=room_name,
+                        room=room,
+                        data=text.replace('\n', '\\n'),
+                    )
+                    room.send_text(text)
+
+                greenlet = self._spawn(_send_global, room_name, message)
+                greenlet.name = (
+                    f'MatrixTransport.send_global '
+                    f'node:{pex(self._raiden_service.address)} '
+                    f'room:{room_name} '
+                    f'user_id:{self._user_id}'
+                )
+            else:
+                gevent.sleep(self._config['retry_interval'])
```

Comment:
It's a bad practice with loops which run while an event is not set to `gevent.sleep` on it, because if the event is set, the sleep doesn't catch it, and can wait for up to `retry_interval` until noticing it should exit. Usually something like `while self._stop_event.wait(self._config['retry_interval']) is not True:` or the ` while True/if: break` equivalent is the best algorithm for this kind of loop.